### ISSUE TITLE
Should we add fields for gene family and gene on rearrangements?

### ISSUE THREAD

**Issue by schristley** [airr-community/airr-standards#258]
The rearrangement schema has locus and allele fields but it does not have fields for the gene family and the gene. For the ADC queries, that means there is no simple mechanism to query on them. The current "hack" is to do a substring search, but this is inefficient, and worse is dependent upon the nomenclature for the name.

**Comment by bcorrie** [comment]
I would support this... We actually have this already, as we already generate these fields internally in our repositories for exactly this purpose. For us to optimize searches, we take v_call and if it is a substring at the gene or family level we do an exact match on our internal ir_gene_family and ir_gene_gene fields 8-)

As discussed, this is error prone as the logic for  mapping a v_call string to a gene or family can be problematic...

**Comment by javh** [comment]
I lean in favor of these being custom fields, as they are gene nomenclature specific concepts.  Having reserved fields for gene and allele might be fine though. Family is more sketchy.

**Comment by schristley** [comment]
> I lean in favor of these being custom fields, as they are gene nomenclature specific concepts.

Yep, I'm mainly thinking about reserving the field names. I consider these to be annotations in the germline_database but that's not going to get resolved anytime quickly.

In the OGRDB [schema](https://github.com/airr-community/ogrdb/blob/master/schema/inferred_gene_submission_schema.yaml#L877) these are the `gene_subgroup` and `subgroup_designation` fields, so instead of specifically saying "family", they are gene groups.

We could just make this an informal spec, or we can have them in the schema.

**Comment by javh** [comment]
`v_allele`, `v_gene`, and `v_subgroup` then?

**Comment by bcorrie** [comment]
Is v_allele == v_call? Or are we saying that this is a shortened field that for IGHV1-1*01 that v_allele == "01"

**Comment by javh** [comment]
I was thinking:

```
v_call: IGHV1-1*01
v_subgroup: IGHV1
v_gene: IGHV1-1
v_allele: 01
```

Which is not perfectly semantically consistent, but matches with the GenBank structure:

```
     V_segment       154..447
                     /gene="IGHV3-73"
                     /inference="COORDINATES: alignment:IgBLAST:1.6.1"
                     /allele="02"
```

**Comment by schristley** [comment]
@javh sounds good, maybe bring up in DataRep call to see if anybody has any comments? What about d, j and c?

**Comment by javh** [comment]
> What about d, j and c?

Same structure, I presume. Though, subgroup and gene will be redundant for the IMGT nomenclature.

**Comment by bcorrie** [comment]
Is there any reason why we don't push this change through with the next release. It doesn't seem controversial to me (no real discussion counter to this) so it seems like maybe we should just do it?

**Comment by javh** [comment]
Seems good to me.

**Comment by schristley** [comment]
Will these be ADC API `required`? I assume we want to support query so `adc-api-optional: true`?

**Comment by bcorrie** [comment]
Supporting query means ```adc-api-optional:false``` doesn't it? Your original post about this is around efficient queries, so I think we want to be able to query on these... I would suggest an x-miairr spec something like this:

```
            x-airr:
                miairr: false
                required: true
                nullable: true
                adc-api-optional: false
```

**Comment by schristley** [comment]
> I would suggest an x-airr spec something like this:

sounds good.

**Comment by bcorrie** [comment]
@williamdlees it was suggested at the last Minimal Standards meeting that we should ask you to review the above changes we are suggesting to the data representation specification, to ensure that we don't do anything the conflicts with the Germline Working Group...

**Comment by bcorrie** [comment]
As per note here (https://github.com/airr-community/airr-standards/pull/295#issuecomment-567555163) we are suggesting pushing the gene naming decision to v2.0 rather than v1.3...

**Comment by bcorrie** [comment]
@schristley can we return this issue to JUST talking about the ADC API extension of having v_subgroup_full and v_gene_full as possible ADC API query fields - and keep that independent of the more general gene naming discussion in #295  

We could then give this a tag ADC API v1.0, agree the API changes, and push them for ADC API v1.0 and not get this API convenience fix embroiled in the germline gene naming discussion...

**Comment by schristley** [comment]
> can we return this issue to JUST talking about the ADC API extension of having v_subgroup_full and v_gene_full as possible ADC API query fields - and keep that independent of the more general gene naming discussion

Well that's all I've been talking about anyways, I wasn't getting into gene nomenclature. But we do have to agree on the content of these fields, and that's been a sticky point...

I think the discussion has opened up a number of issues revolving around the schema and how it gets used by different stakeholders (tools, API, etc.) that haven't been addressed yet, so we should continue that and see where it leads.

**Comment by bcorrie** [comment]
@schristley I think we can close this issue, no? We have #539 to add fields to the ADC query API, and have agreed that we don't want more fields in Rearrangement. So I think this is covered...
